### PR TITLE
Enable nullability in FormAccessibleObject

### PR DESCRIPTION
**Pull request by gpetrou**
## Proposed changes

- Enable nullability in FormAccessibleObject.

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/winforms/pull/6674)

### CODE REVIEW

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs [hunk 1]
```diff
@@ -63,11 +53,6 @@ public override AccessibleRole Role
                         : AccessibleRole.Client;
                 }
             }
-
-            internal override void SetValue(string newValue)
```
Comment (gpetrou) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs:67]: These seem to be the same as in the base, so I removed them.

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs [hunk 2]
```diff
@@ -43,15 +41,7 @@ internal override object GetPropertyValue(UiaCore.UIA propertyID)
...
-            internal override bool IsIAccessibleExSupported()
-            {
-                if (_owner is not null)
-                {
-                    return true;
-                }
-
-                return base.IsIAccessibleExSupported();
-            }
+            internal override bool IsIAccessibleExSupported() => true;
```
Comment (dreddy-work) [inline src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs:44]: nice.  @Tanya-Solyanik / @vladimir-krestov .

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs (hunk 1 above)
Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs:67]: Can you point me where base is defining this? i see empty method there

Comment (gpetrou) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs:67]: https://github.com/dotnet/winforms/blob/main/src/System.Windows.Forms/src/System/Windows/Forms/AccessibleObject.cs#L580

Comment (dreddy-work) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs:67]: Thank you. there was overridable method that confused me.

**Review by RussKie — COMMENTED** [review]
:shipit:

File: src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs (hunk 2 above)
Comment (vladimir-krestov) [reply src/System.Windows.Forms/src/System/Windows/Forms/Form.AccessibleObject.cs:44]: Good change, `_owner` can't be null.

**Review by vladimir-krestov — APPROVED** [review]
LGTM ✔️

**Review by dreddy-work — APPROVED** [review]

**Pull request merged**